SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc

```diff
@@ -439,24 +440,26 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit> > const
     std::vector<PCaloHit> hcalHits;
     hcalHits.reserve(hcalHitsOrig.size());
 
-    for ( unsigned int i=0; i< hcalHitsOrig.size(); i++) {
-      DetId id(hcalHitsOrig[i].id());
-      HcalDetId hid(id);
-      
-      //      if ( id.subdetId()==2 && hid.ietaAbs()<16 ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      if ( !htopoP->validHcal(hid) ) edm::LogError("HcalDigitizer") << "bad hcal id found in digitizer. Skipping " << id.rawId() << std::endl;
-      else
-	hcalHits.push_back(hcalHitsOrig[i]);
-    }
     //evaluate darkening before relabeling
     if(m_HEDarkening || m_HFRecalibration){
-      darkening(hcalHits);
+      darkening(hcalHitsOrig);
```

Comment:
what happens if an invalid detid is passed to darkening()? (as is likely to happen given this code change until someone figures out why we have invalid hcal detids floating around in fastsim (at least))

> On Nov 5, 2015, at 4:37 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> In SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:
> 
> >      //evaluate darkening before relabeling
> >      if(m_HEDarkening || m_HFRecalibration){
> > -      darkening(hcalHits);
> > +      darkening(hcalHitsOrig);
> 
> I don't ever plan to activate the darkening for invalid detids. It only works with HcalTestNumbering enabled. This is specified in the SLHC customizations, and it will be kept when @mark-grimes and I eventually port them to 80X.
> 
> —
> Reply to this email directly or view it on GitHub.
>